sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/factory/credential/AzureAadCredentialBuilderFactory.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.core.factory.credential;
+
+import com.azure.identity.AadCredentialBuilderBase;
+import com.azure.spring.core.aware.AzureProfileAware;
+import com.azure.spring.core.properties.AzureProperties;
+import com.azure.spring.core.properties.PropertyMapper;
+
+import java.util.concurrent.ExecutorService;
+
+/**
+ * A credential builder factory for the {@link com.azure.identity.AadCredentialBuilderBase}.
+ */
+public abstract class AzureAadCredentialBuilderFactory<T extends AadCredentialBuilderBase<T>> extends AbstractAzureCredentialBuilderFactory<T> {
+
+    private ExecutorService executorService = null;
+
+    /**
+     * Create a {@link AzureAadCredentialBuilderFactory} instance with {@link AzureProperties}.
+     * @param azureProperties The Azure properties.
+     */
+    public AzureAadCredentialBuilderFactory(AzureProperties azureProperties) {
+        super(azureProperties);
+    }
+
+    @Override
+    protected void configureService(T builder) {
+        AzureProperties azureProperties = getAzureProperties();
+        AzureProfileAware.Profile profile = azureProperties.getProfile();
+        PropertyMapper map = new PropertyMapper();
+        map.from(azureProperties.getCredential().getClientId()).to(builder::clientId);
```

Comment:
Is it necessary to judge that the client id is empty? if empty, then get `getManagedIdentityClientId()` to assign the client id?

Reply:
Let's keep it this way, but as we discussed this morning, the managed identity client id will probably get removed https://github.com/Azure/azure-sdk-for-java/issues/26953.